models/public/forward_tacotron/description/forward_to_onnx.py

```diff
@@ -0,0 +1,206 @@
+from utils import hparams as hp
+from utils.text.symbols import symbols
+from utils.paths import Paths
+import argparse
+from utils.text import text_to_sequence
+import numpy as np
+
+##################################################################################################
+
+class CBHGWrapper(nn.Module):
+    """Class for changing forward pass in CBHG layer: MaxPool1d to MaxPool2d."""
+    def __init__(self, model):
+        super().__init__()
+
+        # List of all rnns to call `flatten_parameters()` on
+        self.cbhg = model
+        self.maxpool = nn.MaxPool2d(kernel_size=(2, 1), stride=1, padding=(1, 0))
+
+    def forward(self, x):
+        # Save these for later
+        residual = x
+        seq_len = x.size(-1)
+        conv_bank = []
+
+        # Convolution Bank
+        for conv in self.cbhg.conv1d_bank:
+            c = conv(x) # Convolution
+            conv_bank.append(c[:, :, :seq_len])
+
+        # Stack along the channel axis
+        conv_bank = torch.cat(conv_bank, dim=1)
+
+        # dump the last padding to fit residual
+        conv_bank = conv_bank.unsqueeze(-1)
+        x = self.maxpool(conv_bank)
+        x = x.squeeze(-1)
+        x = x[:, :, :seq_len]
+
+        # Conv1d projections
+        x = self.cbhg.conv_project1(x)
+        x = self.cbhg.conv_project2(x)
+
+        # Residual Connect
+        x = x + residual
+
+        # Through the highways
+        x = x.transpose(1, 2)
+        if self.cbhg.highway_mismatch is True:
+            x = self.cbhg.pre_highway(x)
+        for h in self.cbhg.highways: x = h(x)
+
+        x, _ = self.cbhg.rnn(x)
+        return x
+
+
+class DurationPredictor(nn.Module):
+    def __init__(self, model):
+        super().__init__()
+        self.model = model
+        self.model.prenet = CBHGWrapper(self.model.prenet)
+
+    def forward(self, x, alpha=1.0):
+        x = self.model.embedding(x)
+        dur = self.model.dur_pred(x, alpha=alpha)
+
+        x = x.transpose(1, 2)
+        x = self.model.prenet(x)
+
+        return x, dur
+
+    def lr(self, x, dur):
+        return self.model.lr(x, dur)
+
+
+
+
+class Tacotorn(nn.Module):
+    def __init__(self, model):
+        super().__init__()
+        self.model = model
+        self.model.postnet = CBHGWrapper(self.model.postnet)
+
+    def forward(self, x):
+        x, _ = self.model.lstm(x)
+        x = self.model.lin(x)
+        x = x.transpose(1, 2)
+
+        x_post = self.model.postnet(x)
+        x_post = self.model.post_proj(x_post)
+        x_post = x_post.transpose(1, 2)
+        x_post = x_post.squeeze()
+
+        return x_post
+
+
+def main():
+    # Parse Arguments
+    parser = argparse.ArgumentParser(description='TTS Generator')
+
+    parser.add_argument('--tts_weights', type=str, help='[string/path] Load in different FastSpeech weights')
+
+    parser.add_argument('--force_cpu', '-c', action='store_true',
+                        help='Forces CPU-only training, even when in CUDA capable environment')
```

Comment:
Why is this even an option? This script is just to do conversion, there should be no need to use CUDA.